src/timer/mod.rs

```diff
@@ -33,26 +43,58 @@ pub struct TimerResult {
     pub status: ExitStatus,
 }
 
+/// Discard the output of a child process.
+fn discard(output: ChildStdout) {
+    const CHUNK_SIZE: usize = 64 << 10;
```

Comment:
Just out of interest: For `splice`, I assume this needs to be larger than a page size... but performance probably levels off at some point, so there's no gain in making this even larger?

Reply:
`splice()` seems to work for arbitrary sizes, not just pages.  The pipe buffers track byte offsets anyway for `read()` and `write()`.

The default pipe capacity in Linux is 16 pages, so 64K (except for s390x).  For fun, here's some benchmarks:

| `CHUNK_SIZE` | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| 16K | 361.8 ± 12.0 | 338.4 | 374.9 | 1.00 |
| 32K | 341.9 ± 19.1 | 299.0 | 367.3 | 1.00 |
| 64K | 341.9 ± 16.1 | 327.5 | 383.0 | 1.00 |
| 128K | 339.5 ± 14.1 | 309.6 | 356.1 | 1.00 |
| 256K | 339.9 ± 15.8 | 314.5 | 365.8 | 1.00 |

I was about to suggest changing to 128K, but then I checked `strace`:

```
splice(4, NULL, 3, NULL, 131072, 0)     = 65536
splice(4, NULL, 3, NULL, 131072, 0)     = 65536
splice(4, NULL, 3, NULL, 131072, 0)     = 65536
...
```

We can only splice 64K at a time anyway, since that's the capacity of the pipe, so bigger chunks are a waste.

Reply:
Thank you.